--- prova_2.c	2022-05-15 15:36:26.168196300 +0200
+++ prova_1.c	2022-05-15 15:36:19.987302600 +0200
@@ -3,6 +3,7 @@
 
   int i, n;
 
+  // initialize first and second terms
   int t1 = 0, t2 = 1;
 
   // initialize the next term (3rd term)
@@ -14,19 +15,14 @@
 
   // print the first two terms t1 and t2
   printf("Fibonacci Series: %d, %d, ", t1, t2);
-  printf("Fibonacci Series: %d, %d, ", t1, t2);
-  printf("Fibonacci Series: %d, %d, ", t1, t2);
 
   // print 3rd to nth terms
   for (i = 3; i <= n; ++i) {
+    printf("%d, ", nextTerm);
     t1 = t2;
     t2 = nextTerm;
     nextTerm = t1 + t2;
   }
 
-
-
-
-  printf("Fibonacci Series: %d, %d, ", t1, t2);
   return 0;
 }
