--- C:/Users/migue/Desktop/Universita/Magistrale/2_anno/Tesi/Tesi_Magistrale/list_vulnerabilities/Vulnerabilita 6/linux-14132690860e4d06aa3e1c4d7d8e9866ba7756dd/drivers/isdn/capi/kcapi.c
+++ C:/Users/migue/Desktop/Universita/Magistrale/2_anno/Tesi/Tesi_Magistrale/list_vulnerabilities/Vulnerabilita 6/linux-1f3e2e97c003f80c4b087092b225c8787ff91e4d/drivers/isdn/capi/kcapi.c
@@ -479,6 +479,11 @@
 	mutex_lock(&capi_controller_lock);
 
 	ctr_down(ctr, CAPI_CTR_DETACHED);
+
+	if (ctr->cnr < 1 || ctr->cnr - 1 >= CAPI_MAXCONTR) {
+		err = -EINVAL;
+		goto unlock_out;
+	}
 
 	if (capi_controller[ctr->cnr - 1] != ctr) {
 		err = -EINVAL;
